File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<vector<int>, vector<string>>mapr;
        vector <int> count(26,0);
        vector<vector<string>> res;
        
        for(string str: strs){
            fill(count.begin(), count.end(), 0);
            for(char chr: str){
                count[(int)chr - int('a')] += 1;
            }
            mapr[count].push_back(str);
        }
        
        for(auto pair: mapr){
            res.push_back(pair.second);
        }
        
        return res;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        arrs = collections.defaultdict(list)

        for s in strs:
            countr = [0] * 26
            for c in s:
                countr[ord(c) - ord('a')] += 1
            arrs[tuple(countr)].append(s)
        
        return list(arrs.values())